src/driver/hermes_mod.h

```diff
@@ -79,6 +89,47 @@ struct __attribute__((__packed__)) hermes_cfg {
 	uint32_t ehdssze;
 };
 
+struct __attribute__((__packed__)) hermes_cmd_req {
+	uint8_t opcode;
+	uint8_t rsv0;
+	uint16_t cid;
+	uint32_t rsv1;
+	union {
+		struct __attribute__((__packed__)) {
+			uint8_t prog_slot;
+			uint8_t data_slot;
+		} run_prog;
+
+		uint8_t cmd_specific[24];
+	};
+};
+
+struct __attribute__((__packed__)) hermes_cmd_res {
+	uint16_t cid;
+	uint8_t status;
+	uint8_t rsv0[5];
+	union {
+		struct __attribute__((__packed__)) {
+			uint64_t ebpf_ret;
+		} run_prog;
+
+		uint8_t cmd_specific[8];
+	};
+};
+
+struct __attribute__((__packed__)) hermes_cmd {
+	struct hermes_cmd_req req;
+	struct hermes_cmd_res res;
+};
+
+union __attribute__((__packed__)) hermes_cmd_ctrl {
+	struct {
+		uint8_t ehcmdexec:1;
+		uint8_t ehcmddone:1;
```

Comment:
I would avoid bit fields like the plague especially for hardware interfaces. Nothing guarantees the bit order.